accepted/00101-health-check-tool.md

```diff
@@ -0,0 +1,282 @@
+
+### Phase 0 (before 5.0 GA):
+
+#### Uyuni Health Check
+- Create RPM packages in OBS/IBS.
+- Use container images building on OBS/IBS for the different components.
+- Provide an "standalone" version only (no integration with Monitoring stack). Via Uyuni server channel (or others so it is available for engineering or supporters).
+- Flag this as a Tech Preview.
+
+#### Saline
+- Research about using TCP instead of IPC sockets for Salt Master internal sockets.
+- Build Saline in OBS/IBS and provide it via RPM package for the Uyuni Server channel.
+- Saline runs inside "uyuni-server" container.
+
+### Phase 1 (after 5.0 GA):
+
+#### Uyuni Health Check
+- Enhance metrics, dashboards and alerts.
+- Allow user to enhance Loki pipeline to add filters and/or replace patterns to replace sensitive information.
+- Integration with Monitoring Stack. Enhance monitoring formula to enable Health Check monitoring.
+- Reuse configuration from "standalone" version.
+
+#### Saline
+- Provide "Saline" inside a separated container that attach to Salt Master TCP sockets from "uyuni-server" container.
+- Alternatively, explore sharing Salt Master IPC sockets between containers.
+
+
+# How is the exporter going to access to uyuni-server container? For example, to query the database.
+As mentioned, the exporter needs to have access to Uyuni server container. It does need to query the DB and also execute some Salt runners jobs to gather metrics. If the "exporter" runs in a separated container than the Salt Master, then we can make the "exporter" container to run in the same network than the "uyuni-server", then TCP sockets are available to use.
+
+There are some Salt runners jobs that requires access to Salt Master Event publisher, currently exposed via IPC socket. This makes it tricky access these sockets from a different container. Some metrics might not be available until we move to "TCP" sockets for the Salt Master Event publisher in this case.
+
+Alternatively, we could make the "exporter" as part of the server-image, even if not running by default. Then it is up to the Healh Check Tool or Monitoring formula to start it. This would solve the current issue of not having access to the IPC sockets.
+
+
+# Security considerations
+In the current design, we are exposing metrics via Promotheus and making them available in Grafana, and more importantly we are exposing logs messages via Loki/LogCLI to CLI and Grafana users. It is important to notice that after running this tool, and until related containers are destroyed, the Grafana Dashboards (and other components like Prometheus and Loki/LogCLI) are exposing metrics and logs messages that may contain sensitive data and information to any non-root user in the system or to anyone that have access to this host in the network.
+
+Promtail pipeline definition can be enhanced to use [replace](https://grafana.com/docs/loki/latest/send-data/promtail/stages/replace/) function, to hidden sensitive data from logs.
```

Comment:
Will this be configurable in the formula?

Reply:
Yes. Adding a note to clarify this.